docs/website/blog/2023-11-01-dlt-dagster.md

```diff
@@ -0,0 +1,421 @@
+
+### Step 1: Setting up a dagster project:
+
+- Start by creating a new Dagster project scaffold:
+
+```python
+dagster project scaffold --name mongodb-dlt
+```
+
+- Follow the steps mentioned earlier and create an **`assets`**, and **`resources`** directory under **`mongodb-dlt/mongodb_dlt`**.
+- Initialize a **`dlt`** MongoDB pipeline in the same directory:
+
+```python
+dlt init mongodb bigquery
+```
+
+This will create a template with all the necessary logic implemented for extracting data from MongoDB. After running the command your directory structure should be as follows:
+
+```python
+.
+├── README.md
+├── mongodb_dlt
+│   ├── __init__.py
+│   ├── assets
+│   │   ├── __init__.py
+│   │   └── assets.py
+│   ├── mongodb
+│   │   ├── README.md
+│   │   ├── __init__.py
+│   │   └── helpers.py
+│   ├── mongodb_pipeline.py
+│   ├── requirements.txt
+│   └── resources
+│       ├── __init__.py
+├── mongodb_dlt_tests
+│   ├── __init__.py
+│   └── test_assets.py
+├── pyproject.toml
+├── setup.cfg
+└── setup.py
+```
+
+### Step 2: Configuring MongoDB Atlas and Credentials
+
+For this example, we are using MongoDB Atlas. Set up the account for MongoDB Atlas and use the test [Movie Flix Dataset](https://www.mongodb.com/docs/atlas/sample-data/sample-mflix/). You can find detailed information on setting up the credentials in the MongoDB verified sources [documentation](https://dlthub.com/docs/dlt-ecosystem/verified-sources/mongodb).
+
+Next, create a `.env` file and add the BigQuery and MongoDB credentials to the file. The `.env` file should reside in the root directory.
+
+
+### Step 3: Adding the DltResource
+
+ Create a **`DltResouce`** under the **resources** directory. Add the following code to the **`__init__.py` :**
+
+```python
+from dagster import ConfigurableResource 
+
+import dlt
+
+class DltResource(ConfigurableResource):
+    pipeline_name: str
+    dataset_name: str
+    destination: str
+
+    def load_collection(self, resource_data, database):
+
+        # configure the pipeline with your destination details
+        pipeline = dlt.pipeline(
+        pipeline_name=f"{database}_{self.pipeline_name}", destination=self.destination, dataset_name=f"{self.dataset_name}_{database}"
+        )
+
+        load_info = pipeline.run(resource_data, write_disposition="replace")
+
+        return load_info
+```
+
+### Step 4: Defining an Asset Factory
+
+The structure of data in MongoDB is such that under each database you will find multiple collections. When writing a data pipeline it is important to separate the data loading for each collection. 
+
+Dagster provides the feature of `**@multi_asset`** declaration that will allow us to convert each collection under a database into a separate asset. This will make our pipeline easy to debug in case of failure and the collections independent of each other.
```

Comment:
Here and further there is a conflict between quotation marks and asterisks `**@multi_asset`**. Always read the article carefully before publishing using the preview deploy. if you don't know how to do that let me know